src/rxns/rxn_condensed_phase_photolysis.c

```diff
@@ -0,0 +1,486 @@
+    }
+
+    // Add dependence on aerosol-phase water for reactants and products
+    for (int i_react_dep = i_phase * NUM_REACT_;
+         i_react_dep < (i_phase + 1) * NUM_REACT_; i_react_dep++)
+      if (WATER_(i_phase) >= 0) {
+        JAC_ID_(i_jac++) =
+            jacobian_get_element_id(jac, REACT_(i_react_dep), WATER_(i_phase));
+      } else {
+        JAC_ID_(i_jac++) = -1;
+      }
+    for (int i_prod_dep = i_phase * NUM_PROD_;
+         i_prod_dep < (i_phase + 1) * NUM_PROD_; i_prod_dep++)
+      if (WATER_(i_phase) >= 0) {
+        JAC_ID_(i_jac++) =
+            jacobian_get_element_id(jac, PROD_(i_prod_dep), WATER_(i_phase));
+      } else {
+        JAC_ID_(i_jac++) = -1;
+      }
+  }
+
+  return;
+}
+
+/** \brief Update reaction data
+ *
+ * Photolysis reactions can have their base (pre-scaling) rate constants updated
+ * from the host model based on the calculations of an external photolysis
+ * module. The structure of the update data is:
+ *
+ *  - \b int photo_id (Id of one or more photolysis reactions set by the host
+ *       model using the camp_rxn_photolysis::rxn_photolysis_t::set_photo_id
+ *       function prior to initializing the solver.)
+ *  - \b double rate_const (New pre-scaling rate constant.)
+ *
+ * \param update_data Pointer to the updated reaction data
+ * \param rxn_int_data Pointer to the reaction integer data
+ * \param rxn_float_data Pointer to the reaction floating-point data
+ * \param rxn_env_data Pointer to the environment-dependent data
+ * \return Flag indicating whether this is the reaction to update
+ */
+bool rxn_condensed_phase_photolysis_update_data(void *update_data, int *rxn_int_data,
+                                double *rxn_float_data, double *rxn_env_data) {
+  int *int_data = rxn_int_data;
+  double *float_data = rxn_float_data;
+
+  int *photo_id = (int *)update_data;
+  double *base_rate = (double *)&(photo_id[1]);
+
+  // Set the base photolysis rate constants for matching reactions
+  if (*photo_id == RXN_ID_ && RXN_ID_ > 0) {
+    BASE_RATE_ = (double)*base_rate;
+    RATE_CONSTANT_ = SCALING_ * BASE_RATE_;
+    return true;
+  }
+
+  return false;
+}
+
+/** \brief Update reaction data for new environmental conditions
+ *
+ * For Condensed Phase photolysis reaction this only involves recalculating the
+ * forward rate constant.
+ *
+ * \param model_data Pointer to the model data
+ * \param rxn_int_data Pointer to the reaction integer data
+ * \param rxn_float_data Pointer to the reaction floating-point data
+ * \param rxn_env_data Pointer to the environment-dependent parameters
+ */
+void rxn_condensed_phase_photolysis_update_env_state(ModelData *model_data,
+                                                    int *rxn_int_data,
+                                                    double *rxn_float_data,
+                                                    double *rxn_env_data) {
+  // TODO: set the rate from a host model
+  return;
```

Comment:
@mattldawson I'm not sure what I was supposed to do with this. Do you have any idea if this is needed?